### PR TITLE
lab 4

### PR DESCRIPTION
**Pull request by niksone**

### CODE REVIEW

File: lab4/src/map.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    def simulate_euro_diffusion(self) -> None:
+        if len(self.countries) == 1:
+            self.countries[0].day_of_full = 0
+            return
+
+        full = False
+        day = 1
+        while not full:
+            for x in range(10 + 1):
+                for y in range(10 + 1):
```
Comment (dmitry9933) [inline lab4/src/map.py]: Remove "magic" numbers, move them to separate variables.

File: lab4/src/map.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            for city in row:
+                if city is not None:
+                    neighbours_list = self.__get_neighbours(city.x, city.y)
+                    city.set_neighbours(neighbours_list)
+
+    def __get_neighbours(self, x, y) -> List[City]:
+        neighbours = []
+        if self.grid[x][y + 1] is not None:
+            neighbours.append(self.grid[x][y + 1])
+        if self.grid[x][y] is not None:
```
Comment (dmitry9933) [inline lab4/src/map.py]: You made wrong y index, it should be [x][y - 1]

File: lab4/src/input.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        counties_len = int(lines[line_index])
+        line_index += 1
+        countries_list = []
+        for j in range(counties_len):
+            parsed = parse_country(lines[line_index])
+            countries_list.append(parsed)
+            line_index += 1
+        case += 1
+        cases.append(countries_list)
+    return cases
```
Comment (dmitry9933) [inline lab4/src/input.py:41]: Bad readability, use linter

File: lab4/src/map.py (hunk 2 above)
Comment (niksone) [reply lab4/src/map.py]: Good Catch! fixed

File: lab4/src/input.py (hunk 3 above)
Comment (niksone) [reply lab4/src/input.py:41]: Agree! Fixed

File: lab4/src/map.py (hunk 1 above)
Comment (niksone) [reply lab4/src/map.py]: Move to private variable _grid_offset

**Comment by dmitry9933** [comment]
Good!

**Pull request merged**